k9mail-library/src/main/java/com/fsck/k9/mail/autoconfiguration/AutoConfigurationUtil.java

```diff
@@ -0,0 +1,206 @@
+package com.fsck.k9.mail.autoconfiguration;
+
+
+import java.net.UnknownHostException;
+import java.util.Arrays;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.xbill.DNS.DClass;
+import org.xbill.DNS.Lookup;
+import org.xbill.DNS.MXRecord;
+import org.xbill.DNS.Record;
+import org.xbill.DNS.Resolver;
+import org.xbill.DNS.SRVRecord;
+import org.xbill.DNS.SimpleResolver;
+import org.xbill.DNS.TextParseException;
+import org.xbill.DNS.Type;
+
+
+/**
+ * Created by daquexian on 6/9/17.
+ * Util class for autoconfiguration
+ */
+
+public class AutoConfigurationUtil {
+    public static ProviderInfo findProviderMozilla(String domain) {
+        ProviderInfo providerInfo = findProviderInISPDB(domain);
+
+        if (providerInfo != null) {
+            return providerInfo;
+        }
+
+        try {
+            MXRecord mxRecord = mxLookup(domain);
+            if (mxRecord != null) {
+                final String target = mxRecord.getTarget().toString(true);
+                final String[] targetParts = target.split("\\.");
+
+                String newDomain = targetParts[targetParts.length - 2] + "." + targetParts[targetParts.length - 1];
+
+                if (!newDomain.equals(domain)) {
+                    providerInfo = findProviderInISPDB(newDomain);
+                }
+            }
+        } catch (Exception e) {
+
```

Comment:
Add logging for this, especially because you're catching `Exception` rather than something specific.